toolkit/tools/pkg/imagecustomizerlib/customizeboot.go

```diff
@@ -107,143 +93,228 @@ func replaceToken(inputGrubCfgContent string, oldToken string, newToken string)
 	return outputGrubCfgContent, nil
 }
 
-func setLinuxPath(inputGrubCfgContent string, linuxPath string) (outputGrubCfgContent string, oldKernelPath string, err error) {
-	match := linuxPathRegex.FindStringSubmatchIndex(inputGrubCfgContent)
-	if match == nil {
-		return "", "", fmt.Errorf("failed to find kernel command line")
+// Find the linux command within the grub config file.
+func findLinuxLine(inputGrubCfgContent string) ([]grub.Token, error) {
+	linuxLine, err := findSingularGrubCommand(inputGrubCfgContent, "linux")
```

Comment:
how does this handle the case where there are if statements/multiple boot menu items where each has its own linux/initrd commands?

Reply:
It errors out.

If a user ever requests that we handle grub.cfg files with multiple boot entries, then we can figure out how to handle the situation.